src/plugins/data_views/server/plugin.ts

```diff
@@ -53,13 +53,14 @@ export class DataViewsServerPlugin
   }
 
   public start(
-    { uiSettings }: CoreStart,
+    { uiSettings, capabilities }: CoreStart,
     { fieldFormats }: DataViewsServerPluginStartDependencies
   ) {
     const serviceFactory = dataViewsServiceFactory({
       logger: this.logger.get('indexPatterns'),
       uiSettings,
       fieldFormats,
+      capabilities,
```

Comment:
The data view service exposed to other plugins will not have the ability to save since the request is not passed in.

It makes me wonder: will internal plugins need the ability to save? If they do, is there a different way to instantiate a service that can do that?

Reply:
Nevermind, the rest of the code changes answer this question.